RoboJackets/robocup-software
# Make planner node an action server

## Description
Turns planner node into an Action Server.

## Associated / Resolved Issue
 [ClickUp card](https://app.clickup.com/t/2zx11wp)

## Design Documents
[Link](https://docs.google.com/document/d/1qnf3ca5nMZWkTeyERrk6fsng_ZnlourK01f8Z9vLoPs/edit?usp=sharing)

## Steps to Test
### Test Case 1
1. Run sim.
2. Observe Robot 2 heading to point and heading [given by the ActionClient](https://github.com/RoboJackets/robocup-software/pull/1948/files#diff-6ca09102ecda0a74b48b3688563a3f61800650f3cd6c962190122cb83fe9d12eR38-R45) (client will be rewritten in full by @kfu02 )
3. Ensure console outputs is_done when done.

**Expected result:** robot 2 avoids obstacles when going to its point (but only the first time).

## Key Files to Review
Action Server / Client
 * planner_node.*pp
 * robot_action_client.*pp

## Review Checklist
